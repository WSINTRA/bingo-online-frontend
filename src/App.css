.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 20vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}
.Game-layout {
  display: grid;
  grid-template-columns: repeat(3, auto);
  grid-template-rows: repeat(3, auto);
  border-style: solid;
}
.Ball-container {
   position: relative;
  text-align: center;
  color: white;
}
.centered {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  font-size: 10rem;
}
.Player-books {
  border-style: dotted;
}
.Player-controls{
  border-style: dotted;
}
.Player-list {
  border-style: dotted;
}
.Bingo-ball {
  border-style: dotted;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
