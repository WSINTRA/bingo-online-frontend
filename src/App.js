import React from 'react';
import logo from './logo.svg';
import './App.css';
import '../node_modules/bootstrap/dist/css/bootstrap.min.css';
import BigBingoBall from './components/BigBingoBall';
import AllPlayersInGame from './components/AllPlayersInGame';
import PlayerControls from './components/PlayerControls';
import BingoBooks from './components/BingoBooks';
import PlayedBalls from './components/PlayedBalls'

class App extends React.Component {

state = {
  player: {name:"Willy", wins: 23,},
  allPlayers: []
}
componentDidMount() {
  this.setState({
    player: {...this.props.user}
  })
}


  render(){
     return (
    <div className="App">
      <header id="welcome-header" className="App-header">
        Willy's Bingo Fun, welcome {this.state.player.name}<br/> since playing you have had {this.state.player.wins} wins
        <button onClick={()=>this.props.logout()} type="submit" className="btn btn-primary btn-block">logout</button>
        </header>
        <div className="Game-layout">
        <BigBingoBall/>
        <AllPlayersInGame allPlayers={this.state.allPlayers}/>
        <PlayerControls/>
        <BingoBooks/>
        <PlayedBalls/>
        </div>
    
      
    </div>
  );
  }
 
}

export default App;
