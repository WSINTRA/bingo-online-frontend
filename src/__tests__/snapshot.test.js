import React from 'react';
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import App from '../App.js';

afterEach(cleanup);

it("matches snapshot", () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});

it("welcomes the logged in user", ()=>{
	const {welcomeText} = render(<App/>);
	const welcomeHeader = getTestById('welcome-header');
	expect(welcomeHeader.includes())
})