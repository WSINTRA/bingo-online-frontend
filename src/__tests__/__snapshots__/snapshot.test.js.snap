// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      Willy's Bingo Fun, welcome 
      <br />
       since playing you have had  wins
      <button
        class="btn btn-primary btn-block"
        type="submit"
      >
        logout
      </button>
    </header>
    <div
      class="Game-layout"
    >
      <div
        class="Bingo-ball"
      >
        <div
          class="Ball-container"
        >
          <img
            alt=""
            src="bingoBall.png"
          />
          <div
            class="centered"
          >
            23
          </div>
        </div>
        <h3>
          Two and three, twenty three
        </h3>
      </div>
      <div
        class="Player-list"
      >
        <h1>
          CURRENT PLAYERS
        </h1>
      </div>
      <div
        class="Player-controls"
      >
        <h1>
          CONTROLS
        </h1>
      </div>
      <div
        class="Player-books"
      >
        <h1>
          THIS COMPONENT WILL HOUSE THE BINGO BOOKS EVENTUALLY
        </h1>
      </div>
      <div
        class="Played-balls"
      >
        <h1>
          THIS COMPONENT WILL HOUSE THE BALLS THAT HAVE BEEN PLAYED
        </h1>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
